# Taller 1

## Integrantes:

* Juan Sebastián Libreros García
* Pablo Guzmán

# Contexto

Usted hace parte del equipo de analítica de datos del Sunderland AFC de la segunda división inglesa. Se les ha encomendado trabajar en un proyecto sobre reclutar jugadores (scouting). Para ello, cuenta con una base de datos del videojuego FIFA que cuenta con información real de 22,080 jugadores a nivel mundial, recolectada en febrero de 2023. Esta información es confiable dado que dicha empresa cuenta con más de 6 mil scouts alrededor del mundo quienes realizan permanentes evaluaciones sobre el desempeño de los jugadores.

# Problema 

Su jefe le ha pedido una descripción de la base de datos, para conocer mejor cómo pueden trabajar con ella. La base de datos cuenta con una pestaña donde está el diccionario de variables. Tenga en cuenta esta información para poder entender los datos.

In [1]:
import pandas as pd
import matplotlib as plt 
import numpy as np

# Desarrollo

In [2]:
players = pd.read_csv('base_taller.csv', sep=';')

In [7]:
players.head()

player_id                     player_name  age    country  height_cm  \
0     231747            Kylian Mbappé Lottin   23     France        182   
1     192985                 Kevin De Bruyne   31    Belgium        181   
2     188545              Robert Lewandowski   33     Poland        185   
3     165153                   Karim Benzema   34     France        185   
4     158023  Lionel Andrés Messi Cuccittini   35  Argentina        169   

   weight_lbs position  overall  potential Value_millions pref_foot  \
0         161       ST       91         95          190,5     Right   
1         165       CM       91         91          107,5     Right   
2         179       ST       91         91             84     Right   
3         179       CF       91         91             64     Right   
4         148       RW       91         91             54      Left   

   weak_foot  skill_moves  international_reputation Attacking_work_rate  \
0          4            5                         4                High   
1          5            4                         4                High   
2          4            4                         5                High   
3          4            4                         4              Medium   
4          4            4                         5                 Low   

  Defensive_work_rate         body_type club_country            club_name  \
0                 Low            Unique       France  Paris Saint Germain   
1              Medium            Unique      England      Manchester City   
2              Medium            Unique        Spain         Fc Barcelona   
3              Medium  Normal (170-185)        Spain          Real Madrid   
4                 Low            Unique       France  Paris Saint Germain   

   club_overall  
0            84  
1            85  
2            83  
3            85  
4            84

In [14]:
players.isna().sum()

player_id                   0
player_name                 0
age                         0
country                     0
height_cm                   0
weight_lbs                  0
position                    0
overall                     0
potential                   0
Value_millions              0
pref_foot                   0
weak_foot                   0
skill_moves                 0
international_reputation    0
Attacking_work_rate         0
Defensive_work_rate         0
body_type                   0
club_country                0
club_name                   0
club_overall                0
dtype: int64

#### 1. ¿Qué tipo de variable es cada una de las columnas? (recuerde responder según las 4 clasificaciones vistas en clase).

In [3]:
players.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22080 entries, 0 to 22079
Data columns (total 20 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   player_id                 22080 non-null  int64 
 1   player_name               22080 non-null  object
 2   age                       22080 non-null  int64 
 3   country                   22080 non-null  object
 4   height_cm                 22080 non-null  int64 
 5   weight_lbs                22080 non-null  int64 
 6   position                  22080 non-null  object
 7   overall                   22080 non-null  int64 
 8   potential                 22080 non-null  int64 
 9   Value_millions            22080 non-null  object
 10  pref_foot                 22080 non-null  object
 11  weak_foot                 22080 non-null  int64 
 12  skill_moves               22080 non-null  int64 
 13  international_reputation  22080 non-null  int64 
 14  Attacking_work_rate   

| Atributo                    | Tipo de variable                | Justificación                                                                 |
|----------------------------|----------------------------------|-------------------------------------------------------------------------------|
| player_id                  | Numérica discreta                | Identificador único, valores enteros sin decimales                            |
| player_name                | Categórica nominal               | Nombres de jugadores, no existe orden                                         |
| age                        | Numérica discreta                | Edad en años, valores enteros                                                 |
| country                    | Categórica nominal               | País de origen, no existe orden                                               |
| height_cm                  | Numérica continua                | Estatura en cm, puede tomar cualquier valor dentro de un rango                |
| weight_lbs                 | Numérica continua                | Peso en libras, puede tomar cualquier valor dentro de un rango                |
| position                   | Categórica nominal               | Posición en el campo, no existe orden                                         |
| overall                    | Numérica discreta                | Puntaje entero de desempeño                                                   |
| potential                  | Numérica discreta                | Puntaje entero de potencial                                                   |
| Value_millions             | Numérica continua                | Valor en millones, puede tomar decimales                                      |
| pref_foot                  | Categórica nominal               | Pie preferido, no existe orden                                                |
| weak_foot                  | Numérica discreta                | Puntaje entero de habilidad con pie débil                                     |
| skill_moves                | Numérica discreta                | Puntaje entero de habilidad técnica                                           |
| international_reputation   | Numérica discreta                | Puntaje entero de reputación internacional                                    |
| Attacking_work_rate        | Categórica ordinal               | Nivel de esfuerzo ofensivo (Low, Medium, High), existe orden                  |
| Defensive_work_rate        | Categórica ordinal               | Nivel de esfuerzo defensivo (Low, Medium, High), existe orden                 |
| body_type                  | Categórica nominal               | Tipo de cuerpo, no existe orden                                               |
| club_country               | Categórica nominal               | País del club, no existe orden                                                |
| club_name                  | Categórica nominal               | Nombre del club, no existe orden                                              |
| club_overall               | Numérica discreta                | Puntaje entero de desempeño del club                                          |

---------------------------------------------------------------------------------------------------------------------------------------

#### 2. Realice un completo análisis descriptivo de las columnas que vamos a utilizar:

a) Age
b) height_cm
c) position
d) club_country
e) overall
f) club_overall
g) value_millions
h) pref_foot

-----------------------------------------------------------------------------------

#### 3. Un miembro del cuerpo técnico asegura que los arqueros (GK) altos tienen un mejor desempeño (overall). ¿Qué puede afirmar usted a través de un análisis descriptivo? (Nota: tome por definición tres grupos de estatura: i) menor a 185 cm; ii) entre 185 y 192 cm; iii) más de 192 cm)

In [12]:
goalKeepers = players[players['position'] == 'GK']
goalKeepers = goalKeepers[['position', 'overall', 'height_cm',]]
goalKeepers.head(10)

position  overall  height_cm
6        GK       90        199
7        GK       89        193
11       GK       89        188
14       GK       89        193
16       GK       88        196
18       GK       88        188
22       GK       88        187
28       GK       87        191
33       GK       87        192
37       GK       85        195

In [13]:
goalKeepers.describe()

overall    height_cm
count  2430.000000  2430.000000
mean     63.899588   188.593827
std       7.449501     4.758131
min      46.000000   165.000000
25%      59.000000   185.000000
50%      64.000000   188.000000
75%      69.000000   192.000000
max      90.000000   206.000000

--------------------------------------------------------------------------------------

#### 4. Un analista afirma que los jugadores zurdos, tienen un mejor desempeño (overall) que los diestros. ¿Qué puede decir sobre esta afirmación a través de un análisis descriptivo?

---------------------------------------------------------------------------------------------------

#### 5. Un scout asegura que los mejores jugadores, según el puntaje de desempeño (overall), juegan en clubes que pertenecen al grupo de “ligas top5” (Inglaterra, España, Alemania, Italia y Francia). ¿Qué puede decir sobre estas dos afirmaciones a través de un análisis descriptivo?

----------------------------------------------------------------------------------------

#### 6. Otro scout asegura que la proporción de jugadores zurdos es mayor en la posición de centro campista (CM) que en la de centro campista ofensivo (CAM). ¿Qué puede decir sobre esta afirmación?

-------------------------------------------------------------------------------------------------------------------

#### 7. Otro scout asegura que los mejores jugadores juegan en los mejores clubes del mundo ¿Es cierta esta afirmación de que existe una correlación entre el desempeño de los jugadores (overall) y el puntaje de desempeño del club al que pertenece (club_overall)?